INSTALL.md

```diff
@@ -156,3 +156,22 @@ The GUI depends on the `capstone`, `freetype` and `glfw` libraries and their hea
 
     # On MacOS
     $ brew install capstone freetype2 glfw
+
+## Building with Rust
+
+Configuring with `--enable-rust` will build and embed components written in the [Rust](https://rust-lang.org) programming language. These components are currently optional and experimental.
```

Comment:
Are there `CFLAGS` equivalent that need to be passed in/initialized so that symbols etc are created properly?

Reply:
I'm not sure what you mean by symbols being created, but you shouldn't have to adjust CFLAGS.

None of this is wired up on windows yet at all. I can try that now if you like?

Reply:
what I meant is that we have default optimization flags for C/C++ and I don't know how Rust configures those things, I imagine that there is some equivalent

Reply:
re: Windows, we'll have to do that as some point yes but I don't think we need to block on it for this PR.

Now that I think of it (related):
we should remove the Github action stuff for now and we should make the appropriate changes when we actually take the Rust dependency (either for all builds or just for "protocol next" depending on where we'll start bringing in Rust), and when we do this we'll want to have Windows figured out for sure (to avoid having to manage many targets - right now Windows only allows to build "protocol next" for that reason).

Reply:
Added a windows sketch which gets far enough along that it passes the round-trip tests between C++ and rust. It's a little ugly and I could use some help with it, I think it's probably not 100% right structurally, I think it rebuilds too often, and also even if working .. it has some limitations (eg. can't make a debug profile that uses it, as rust and/or cargo can only use the release CRT when linking statically; maybe the trick is to go dynamic in the debug config?)

I'll remove the github actions stuff tomorrow I guess. I am a little hesitant to do so since it'll bitrot and break if not checked, but I guess we're not _quite_ ready to turn it on yet.

Reply:
Thanks for @MonsieurNicolas finding [this github comment](https://github.com/rust-lang/rust/issues/39016?msclkid=6b0248f6c5ac11ecbd155fd40b4d4f30#issuecomment-853964918) the debug build now also works on windows.

Currently there's only one Rust source file (the crate root) so the CustomBuild step can just include it as its AdditionalInputs source; this will get unwieldy if we want visual studio / msbuild to re-run cargo on any edit to _any_ Rust file. There's an even-more-involved technique we can use for this [explained here](https://www.reedbeta.com/blog/custom-toolchain-with-msbuild/) but it's not really clear to me that we're going to be doing edit-and-rebuild of Rust code from inside visual studio -- it doesn't have a Rust editing component currently, only does very modest syntax highlighting. No intellisense or anything, you have to use a different IDE for that. I can investigate that custom toolchain approach if we want to go further in that direction, but I think for now it's probably good enough to land.